solver/llbsolver/solver.go

```diff
@@ -106,6 +106,7 @@ func (s *Solver) Solve(ctx context.Context, id string, req frontend.SolveRequest
 	var res *frontend.Result
 	if s.gatewayForwarder != nil && req.Definition == nil && req.Frontend == "" {
 		fwd := gateway.NewBridgeForwarder(ctx, s.Bridge(j), s.workerController)
+		defer fwd.Discard()
```

Comment:
Does this now call `Discard` multiple times? May not be a big issue it `Release()` works correctly (and error seems to be ignored) but would be nicer to explicitly protect it.

Reply:
> Does this now call Discard multiple times? 

I don't think so -- this call and the other (the one in `gatewayFrontend.Solve`) are mutually exclusive, the latter is the one where we `Exec` a frontend which is explicitly not what we do here.

Reply:
To mitigate I will (running `make test` now):
* Ensured that `lbf.refs` is only ever accessed under `lbf.mu` (it already was most of the time, but a couple were missing, I think on code paths which were expected to be non-concurrent)
* Removed non-kept refs from `lbf.refs` after releasing.